cli/src/cli_util.rs

```diff
@@ -1781,6 +1781,26 @@ pub fn print_checkout_stats(ui: &mut Ui, stats: CheckoutStats) -> Result<(), std
     Ok(())
 }
 
+pub fn print_git_import_stats(
+    ui: &mut Ui,
+    tx: &WorkspaceCommandTransaction,
+    stats: &GitImportStats,
+) -> Result<(), CommandError> {
+    // TODO: maybe better to write status messages to stderr
+    if !stats.abandoned_commits.is_empty() {
+        ui.write("Abandoned the following commits:\n")?;
```

Comment:
Perhaps make this say "Import of git refs abandoned the following commits:" for the `jj git import` case? Otherwise, the user will think that `jj log` abandoned their commits.

Reply:
For the fetch, you could perhaps also adjust the message to "Fetch of git refs...".

Reply:
I can't think of a good wording that works for all of the import/fetch/push cases. The automated import_refs() is confusing, but for the other cases, user should know the context.

So I think we can insert additional message if automated import_refs() detected changes.
```
Imported changes from underlying Git repo
Abandoned the following commits:
  kkmpptxz 35605592 master | (empty) bar
Rebased ... descendant commits
Working copy now at: yqosqzyt 096dc80d (empty) (no description set)
Parent commit      : qpvuntsm 230dd059 (empty) (no description set)
...
```

Reply:
Perhaps you could pass a slightly different string to the function at each call site?

>  I think we can insert additional message if automated import_refs() detected changes.

This is an interesting idea, worth trying. It's not immediately clear to me whether this would be so noisy as to be annoying, but there's a good chance it's fine.

Reply:
Though, at a glance I don't see a separate call site for `git push`. :(

Reply:
> insert additional message if automated import_refs() detected changes.

I've made this change. PTAL.

> pass a slightly different string to the function at each call site

It'll get complicated if we add other stats like new/changed local branches.

Reply:
> I've made this change. PTAL.

I'm still worried about this being annoying. I can try using it for a while and see how bad it gets.

Reply:
I'm also still worried about the `push` case. I'm not sure how to make that any clearer without customizing the message for the call site. Fortunately, users shouldn't see that too often.

I didn't follow your thought about how complicated it will get. Superficially, it seems like we're OK now and I hope becoming more vague if necessary in the future might be good enough. But if you think that customization is not an option, we can try this way and see how it goes.

Reply:
> I'm still worried about this being annoying.

It'll be showed only when you've changed the repo by git command, so I hope it wouldn't be too verbose. We have some test changes where external git command isn't involved, but they're probably a bug of `git::export_refs()`.

> I'm also still worried about the push case.

I think the push scenario is fundamentally unintuitive. The user might just want to push the deleted ref (without abandoning the local commit.)

Reply:
> It'll be showed only when you've changed the repo by git command, so I hope it wouldn't be too verbose. 

Yes, I tried it and it's much better than I thought it would be.

I also noticed that this kind of output is easy to miss for me because I have an analogue of `watch jj log` running.

> We have some test changes where external git command isn't involved, but they're probably a bug of git::export_refs().

Interesting.